src/scripts/token.js

```diff
@@ -0,0 +1,74 @@
+export async function fetchWithToken(url, options) {
```

Comment:
Think it would be a great idea to have a more standardized approach to how we are going to communicate with the API. I have just created a PR #19 with a draft.